src/Presentation/Console/Command/QaAutomationCommand.php

```diff
@@ -0,0 +1,171 @@
+use SprykerSdk\Sdk\Core\Application\Dependency\ProjectSettingRepositoryInterface;
+use SprykerSdk\Sdk\Core\Application\Dependency\Repository\TaskRepositoryInterface;
+use SprykerSdk\Sdk\Core\Application\Service\ContextFactory;
+use SprykerSdk\Sdk\Core\Application\Service\ProjectWorkflow;
+use SprykerSdk\Sdk\Core\Application\Service\TaskExecutor;
+use SprykerSdk\Sdk\Core\Domain\Entity\Command;
+use SprykerSdk\Sdk\Core\Domain\Entity\Lifecycle\InitializedEventData;
+use SprykerSdk\Sdk\Core\Domain\Entity\Lifecycle\Lifecycle;
+use SprykerSdk\Sdk\Core\Domain\Entity\Lifecycle\RemovedEventData;
+use SprykerSdk\Sdk\Core\Domain\Entity\Lifecycle\UpdatedEventData;
+use SprykerSdk\Sdk\Core\Domain\Entity\Task;
+use SprykerSdk\SdkContracts\Entity\ContextInterface;
+use SprykerSdk\SdkContracts\Entity\ErrorCommandInterface;
+use SprykerSdk\SdkContracts\Entity\ExecutableCommandInterface;
+use SprykerSdk\SdkContracts\Entity\StagedTaskInterface;
+use SprykerSdk\SdkContracts\Entity\TaskInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class QaAutomationCommand extends RunTaskWrapperCommand
+{
+    /**
+     * @var string
+     */
+    protected const TASKS_SETTING_KEY = 'qa_tasks';
+
+    /**
+     * @var string
+     */
+    protected const DESCRIPTION = 'Run configurable qa tasks.';
+
+    /**
+     * @var string
+     */
+    protected const COMMAND_NAME = 'qa:run';
+
+    /**
+     * @var \SprykerSdk\SdkContracts\Entity\TaskInterface
+     */
+    protected TaskInterface $task;
+
+    /**
+     * @var \SprykerSdk\Sdk\Core\Application\Dependency\Repository\TaskRepositoryInterface
+     */
+    protected TaskRepositoryInterface $taskRepository;
+
+    /**
+     * @param \SprykerSdk\Sdk\Core\Application\Service\TaskExecutor $taskExecutor
+     * @param \SprykerSdk\Sdk\Core\Application\Service\ProjectWorkflow $projectWorkflow
+     * @param \SprykerSdk\Sdk\Core\Application\Dependency\ContextRepositoryInterface $contextRepository
+     * @param \SprykerSdk\Sdk\Core\Application\Dependency\ProjectSettingRepositoryInterface $projectSettingRepository
+     * @param \SprykerSdk\Sdk\Core\Application\Service\ContextFactory $contextFactory
+     * @param \SprykerSdk\Sdk\Presentation\Console\Command\OptionExtractor $optionExtractor
+     * @param \SprykerSdk\Sdk\Core\Application\Dependency\Repository\TaskRepositoryInterface $taskRepository
+     */
+    public function __construct(
+        TaskExecutor $taskExecutor,
+        ProjectWorkflow $projectWorkflow,
+        ContextRepositoryInterface $contextRepository,
+        ProjectSettingRepositoryInterface $projectSettingRepository,
+        ContextFactory $contextFactory,
+        OptionExtractor $optionExtractor,
+        TaskRepositoryInterface $taskRepository
+    ) {
+        $this->taskRepository = $taskRepository;
+        try {
+            $taskIds = $projectSettingRepository->getOneByPath(static::TASKS_SETTING_KEY)->getValues();
```

Comment:
This logic should be moved from the command for reusability and SRP reasons. Please, create a separate class and move logic from the constructor there.

Reply:
needs to think